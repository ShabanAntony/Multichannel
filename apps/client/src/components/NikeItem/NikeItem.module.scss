.base {
  width: 200px;

  &:last-of-type {
    margin-right: auto;
  }

  &__skeleton {
    height: 450px;
    background-color: rebeccapurple;
  }
}

.name {
  font-weight: 700;
}

.image {
  display: inline-block;
  height: 280px;
  margin-bottom: 10px;
  border-radius: 10px;
  overflow: hidden;
  outline: 4px solid white;

  img {
    object-fit: contain !important;
  }

  .base__new & {
    outline-color: red;
  }
}

.discount {
  color: chartreuse;
}

.finalPrice {
  color: red;
  font-weight: 700;
}

.swatches {
  display: flex;
  flex-direction: row;
  flex-wrap: wrap;
  gap: 10px;
  margin: 20px 0;
}

.swatch {
  &_link {
    width: 35px;
    height: 35px;
    border-radius: 50%;
    overflow: hidden;
  }
}
