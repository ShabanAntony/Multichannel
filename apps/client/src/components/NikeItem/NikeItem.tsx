import React, { FC, memo } from 'react'

import { NikeItemEntity } from '@server-api'
import classNames from 'classnames'

import { laryToUSD } from '../../constants/prices'
import {
  calculateNikeFinalPrice,
  formatLaryPrice,
  formatUSDPrice,
} from '../../utils/price'
import { ImageLink } from '../ImageLink/ImageLink'

import styles from './NikeItem.module.scss'

export interface Props {
  readonly data: NikeItemEntity & { discount: number }
}

export const NikeItem: FC<Props> = memo(function NikeItem({
  data: { title, url, image, price, salePrice, discount, isNew },
}) {
  const finalPrice = calculateNikeFinalPrice(salePrice)
  const baseClasses = classNames(styles.base, {
    [styles.base__new]: isNew,
  })

  return (
    <li className={baseClasses}>
      <ImageLink
        className={styles.image}
        link={url}
        src={image}
        alt={title}
        width={200}
        height={280}
      />
      <p className={styles.name}>{title}</p>
      <p>
        Prices: {price}
        {' > '}
        <span className={styles.discount}>
          {salePrice} ({Math.round(discount)}%)
        </span>
        {' > '}
        <span className={styles.finalPrice}>
          {formatUSDPrice(finalPrice)} (
          {formatLaryPrice(finalPrice * laryToUSD)})
        </span>
      </p>
    </li>
  )
})
