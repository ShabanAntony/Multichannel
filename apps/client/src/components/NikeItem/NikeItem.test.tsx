import React from 'react'
import renderer from 'react-test-renderer'

import { NikeItemEntity } from '@server-api'

import { NikeItem } from './NikeItem'

jest.mock('../ImageLink/ImageLink', () => ({
  ImageLink: 'ImageLink',
}))

const mockItem: NikeItemEntity & { discount: number } = {
  discount: 25,
  id: '2',
  image: 'image_2',
  subtitle: 'subtitle_2',
  title: 'title_2',
  url: 'url_2',
  price: 100,
  salePrice: 75,
  isNew: false,
}

describe('NikeItem', () => {
  it('should renders regular item', () => {
    const tree = renderer.create(<NikeItem data={mockItem} />).toJSON()

    expect(tree).toMatchSnapshot()
  })

  it('should renders new item', () => {
    const tree = renderer
      .create(<NikeItem data={{ ...mockItem, isNew: true }} />)
      .toJSON()

    expect(tree).toMatchSnapshot()
  })
})
