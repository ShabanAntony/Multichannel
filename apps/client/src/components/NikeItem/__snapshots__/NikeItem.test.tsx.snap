// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NikeItem should renders new item 1`] = `
<li
  className="base base__new"
>
  <ImageLink
    alt="title_2"
    className="image"
    height={280}
    link="url_2"
    src="image_2"
    width={200}
  />
  <p
    className="name"
  >
    title_2
  </p>
  <p>
    Prices: 
    100
     &gt; 
    <span
      className="discount"
    >
      75
       (
      25
      %)
    </span>
     &gt; 
    <span
      className="finalPrice"
    >
      108$
       (
      292 ლ
      )
    </span>
  </p>
</li>
`;

exports[`NikeItem should renders regular item 1`] = `
<li
  className="base"
>
  <ImageLink
    alt="title_2"
    className="image"
    height={280}
    link="url_2"
    src="image_2"
    width={200}
  />
  <p
    className="name"
  >
    title_2
  </p>
  <p>
    Prices: 
    100
     &gt; 
    <span
      className="discount"
    >
      75
       (
      25
      %)
    </span>
     &gt; 
    <span
      className="finalPrice"
    >
      108$
       (
      292 ლ
      )
    </span>
  </p>
</li>
`;
